### ISSUE TITLE
"variant did not have ANN annotation" in EFFECT field of .tab

### ISSUE THREAD

**Issue by Perugolate** [tseemann/snippy#40]
the "EFFECT" field of the .tab output contains mostly "variant did not have ANN annotation".

snippy v2.9
gbk reference from prokka v1.11

Only ever used snippy with prokka gbk files and never had "variant did not have ANN annotation" before. Tried reannotating with prokka but issue presists.

**Comment by tseemann** [comment]
What version of snpEff are you using?
```
snpEff -version
```
How many SNPs did Snippy find?
Were they in coding regions (ie. CDS features)

**Comment by Perugolate** [comment]
See the log and .tab output below. snpEff is the one bundled with snippy 2.9 (4.1l). I don't have problems with using snippy with other prokka gbk files for any other projects...

.fasta: https://dl.dropboxusercontent.com/u/19097397/pol_3.fna.gz
.gbk: https://dl.dropboxusercontent.com/u/19097397/pol_3.gbk.gz

```
[23:46:43] This is snippy 2.9
[23:46:43] Written by Torsten Seemann <torsten.seemann@gmail.com>
[23:46:43] Obtained from https://github.com/tseemann/snippy
[23:46:43] Detected operating system: linux
[23:46:43] Enabling bundled linux tools.
[23:46:43] Found bwa - /usr/bin/bwa
[23:46:43] Found samtools - /usr/local/bin/samtools
[23:46:43] Found tabix - /usr/bin/tabix
[23:46:43] Found bgzip - /usr/bin/bgzip
[23:46:43] Found snpEff - /home/paul/opt/snippy/bin/../binaries/noarch/snpEff
[23:46:43] Found java - /usr/bin/java
[23:46:43] Found gzip - /bin/gzip
[23:46:43] Found parallel - /home/paul/opt/snippy/bin/../binaries/noarch/parallel
[23:46:43] Found freebayes - /usr/local/bin/freebayes
[23:46:43] Found freebayes-parallel - /home/paul/opt/snippy/bin/../binaries/noarch/freebayes-parallel
[23:46:43] Found fasta_generate_regions.py - /home/paul/opt/snippy/bin/../binaries/noarch/fasta_generate_regions.py
[23:46:43] Found vcfstreamsort - /home/paul/src/freebayes/vcflib/bin//vcfstreamsort
[23:46:43] Found vcfuniq - /home/paul/src/freebayes/vcflib/bin//vcfuniq
[23:46:43] Found vcffirstheader - /home/paul/src/freebayes/vcflib/bin//vcffirstheader
[23:46:43] Found vcf-consensus - /usr/bin/vcf-consensus
[23:46:43] Found snippy-vcf_to_tab - /home/paul/opt/snippy/bin/snippy-vcf_to_tab
[23:46:43] Found snippy-vcf_report - /home/paul/opt/snippy/bin/snippy-vcf_report
[23:46:43] Found snippy-vcf_filter - /home/paul/opt/snippy/bin/snippy-vcf_filter
[23:46:43] Using reference: /media/tera5/olga/an_b1/pol_3.gbk
[23:46:43] Will use 12 CPU cores.
[23:46:43] Using read file: /media/tera5/olga/begendiv/11_S19_L001_R1_001.fastq.gz
[23:46:43] Using read file: /media/tera5/olga/begendiv/11_S19_L001_R2_001.fastq.gz
[23:46:43] Creating folder: 11_S19
[23:46:43] Changing working directory: 11_S19
[23:46:43] Creating reference folder: reference
[23:46:43] Extracting FASTA and GFF from reference.
[23:46:44] Wrote 1 sequences to ref.fa
[23:46:44] Wrote 2511 features to ref.gff
[23:46:44] Creating reference/snpeff.config
[23:46:44] Freebayes will process 48 chunks of 57165 bp, 12 chunks at a time.
[23:46:44] Using BAM RG (Read Group) ID: 11_S19
[23:46:44] Running: samtools faidx reference/ref.fa 2>> 11_S19.log
[23:46:44] Running: bwa index reference/ref.fa 2>> 11_S19.log
[23:46:46] Running: mkdir reference/genomes && cp -f reference/ref.fa reference/genomes/ref.fa 2>> 11_S19.log
[23:46:46] Running: mkdir reference/ref && bgzip -c reference/ref.gff > reference/ref/genes.gff.gz 2>> 11_S19.log
[23:46:46] Running: snpEff build -c reference/snpeff.config -dataDir . -gff3 ref 2>> 11_S19.log
[23:46:49] Running: (bwa mem  -v 2 -M -R '@RG   ID:11_S19       SM:11_S19' -t 12 reference/ref.fa /media/tera5/olga/begendiv/11_S19_L001_R1_001.fastq.gz /media/tera5/olga/begendiv/11_S19_L001_R2_001.fastq.gz | samtools view -@ 12 -F 3844 -q 60 -S -b -u -T ref.fa - | samtools sort -@ 12 - 11_S19) 2>> 11_S19.log
[23:46:54] Running: samtools index 11_S19.bam 2>> 11_S19.log
[23:46:55] Running: samtools depth -q 20 11_S19.bam | bgzip > 11_S19.depth.gz 2>> 11_S19.log
[23:46:56] Running: tabix -s 1 -b 2 -e 2 11_S19.depth.gz 2>> 11_S19.log
[23:46:57] Running: fasta_generate_regions.py reference/ref.fa.fai 57165 > reference/ref.txt 2>> 11_S19.log
[23:46:57] Running: freebayes-parallel reference/ref.txt 12 -p 1 -q 20 -m 60 --min-coverage 10 -V -f reference/ref.fa 11_S19.bam > 11_S19.raw.vcf 2>> 11_S19.log
[23:47:02] Running: /home/paul/opt/snippy/bin/snippy-vcf_filter --minqual 10 --mincov 10 --minfrac 0.9  11_S19.raw.vcf > 11_S19.filt.vcf 2>> 11_S19.log
[23:47:02] Running: snpEff ann -no-downstream -no-upstream -no-intergenic -no-utr -c reference/snpeff.config -dataDir . -noStats ref 11_S19.filt.vcf > 11_S19.vcf 2>> 11_S19.log
[23:47:04] Running: bgzip -c 11_S19.vcf > 11_S19.vcf.gz 2>> 11_S19.log
[23:47:04] Running: tabix -p vcf 11_S19.vcf.gz 2>> 11_S19.log
[23:47:04] Running: /home/paul/opt/snippy/bin/snippy-vcf_to_tab --gff reference/ref.gff --ref reference/ref.fa --vcf 11_S19.vcf > 11_S19.tab 2>> 11_S19.log
[23:47:06] Running: vcf-consensus 11_S19.vcf.gz < reference/ref.fa > 11_S19.consensus.fa 2>> 11_S19.log
[23:47:07] Running: /home/paul/opt/snippy/bin/snippy-vcf_filter --subs 11_S19.filt.vcf > 11_S19.filt.subs.vcf 2>> 11_S19.log
[23:47:07] Running: bgzip -c 11_S19.filt.subs.vcf > 11_S19.filt.subs.vcf.gz 2>> 11_S19.log
[23:47:07] Running: tabix -p vcf 11_S19.filt.subs.vcf.gz 2>> 11_S19.log
[23:47:07] Running: vcf-consensus 11_S19.filt.subs.vcf.gz < reference/ref.fa > 11_S19.consensus.subs.fa 2>> 11_S19.log
[23:47:07] Generating aligned/masked FASTA relative to reference: 11_S19.aligned.fa
[23:47:11] Creating extra output files: BED GFF CSV TXT HTML
[23:47:11] Identified 10 variants.
[23:47:11] Result folder: 11_S19
[23:47:11] Result files:
[23:47:11] * 11_S19/11_S19.aligned.fa
[23:47:11] * 11_S19/11_S19.bam
[23:47:11] * 11_S19/11_S19.bam.bai
[23:47:11] * 11_S19/11_S19.bed
[23:47:11] * 11_S19/11_S19.consensus.fa
[23:47:11] * 11_S19/11_S19.consensus.subs.fa
[23:47:11] * 11_S19/11_S19.csv
[23:47:11] * 11_S19/11_S19.depth.gz
[23:47:11] * 11_S19/11_S19.depth.gz.tbi
[23:47:11] * 11_S19/11_S19.filt.subs.vcf
[23:47:11] * 11_S19/11_S19.filt.subs.vcf.gz
[23:47:11] * 11_S19/11_S19.filt.subs.vcf.gz.tbi
[23:47:11] * 11_S19/11_S19.filt.vcf
[23:47:11] * 11_S19/11_S19.gff
[23:47:11] * 11_S19/11_S19.html
[23:47:11] * 11_S19/11_S19.log
[23:47:11] * 11_S19/11_S19.raw.vcf
[23:47:11] * 11_S19/11_S19.tab
[23:47:11] * 11_S19/11_S19.txt
[23:47:11] * 11_S19/11_S19.vcf
[23:47:11] * 11_S19/11_S19.vcf.gz
[23:47:11] * 11_S19/11_S19.vcf.gz.tbi
[23:47:11] Walltime used: 28 seconds
[23:47:11] May the SNPs be with you.
[23:47:11] Done.
```

```
CHROM   POS     TYPE    REF     ALT     EVIDENCE        FTYPE   STRAND  NT_POS  AA_POS  EFFECT  LOCUS_TAG       GENE    PRODUCT
jla513_unpol    110018  snp     C       A       A:14 C:0        CDS     -       46/957  16/318  missense_variant c.46G>T p.Val16Phe     PROKKA_00090    phnD    Phosphate-import protein PhnD precursor
jla513_unpol    953491  snp     C       G       G:13 C:0        CDS     +                       variant did not have ANN annotation     PROKKA_00877    pchA    Salicylate biosynthesis isochorismate synthase
jla513_unpol    1078712 complex GGTG    CGG     CGG:12 GGTG:0   CDS     -                       variant did not have ANN annotation     PROKKA_00995            hypothetical protein
jla513_unpol    1078726 snp     C       T       T:13 C:0        CDS     -                       variant did not have ANN annotation     PROKKA_00995            hypothetical protein
jla513_unpol    1078735 complex ACGA    GCGG    GCGG:13 ACGA:0  CDS     -                       variant did not have ANN annotation     PROKKA_00995            hypothetical protein
jla513_unpol    1368151 snp     C       T       T:10 C:0        CDS     -                       variant did not have ANN annotation     PROKKA_01257    odhA    2-oxoglutarate dehydrogenase E1 component
jla513_unpol    1596511 ins     TAA     TAAA    TAAA:9 TAA:1    CDS     -                       variant did not have ANN annotation     PROKKA_01460    mnmA_2  tRNA-specific 2-thiouridylase MnmA
jla513_unpol    1946969 snp     A       T       T:13 A:0        CDS     -                       variant did not have ANN annotation     PROKKA_01802    ytrA    HTH-type transcriptional repressor YtrA
jla513_unpol    2010369 snp     G       C       C:12 G:1        CDS     -                       variant did not have ANN annotation     PROKKA_01858    rsbW    Serine-protein kinase RsbW
jla513_unpol    2660349 del     ACCCCCCT        ACCCCCT ACCCCCT:13 ACCCCCCT:0   CDS     -                       variant did not have ANN annotation     PROKKA_02466    hisF    Imidazole glycerol phosphate synthase subunit HisF
```

**Comment by tseemann** [comment]
@Perugolate 

Thank you for sending the GBK file through. I simulated some reads from a mutated version, and ran Snippy with your GBK.

I get effect annotations for all the genes up to position 380847 in the genome but none after it! Hmmm.

So I took the FNA and ran the latest Prokka (github 1.12-dev) on it, then ran the same reads with Snippy on that. It correctly annotated all the exact same variants.

So something weird is happening. Your GBK and the newer GBK are very similar. So my guess is that something has changed in Prokka or `tbl2asn` which generates the GBK files?

Are you able to install the git version of Prokka? 

(your log above shows you are using the last release 1.11 which is quite old and I need to update)

Torsten
